### DISCUSSION TITLE
Insert value or get it

### DISCUSSION THREAD

**Discussion by Distractic** [komapper/komapper#961] [Q&A] [answered]
Hello !

Currently, I have this request :
```kt
@KomapperEntity
@KomapperTable("guild_member")
public data class GuildMember(
    @KomapperId
    val guildId: Int,
    @KomapperId
    val entityId: String,
    @KomapperCreatedAt
    val createdAt: Instant = Instant.EPOCH,
)

override suspend fun addMember(guildId: Int, entityId: String): GuildMember? {
    requireEntityIdNotBlank(entityId)

    val member = GuildMember(guildId, entityId)

    val query = QueryDsl.insert(_GuildMember.guildMember)
        .onDuplicateKeyIgnore()
        .executeAndGet(member)

    return database.runQuery(query)
}
```

My plan is :
* When the keys are not present, a new value will be inserted
* When the keys are present (conflict), the value is retrieved

The problem is that the method `onDuplicateKeyIgnore` will return `null` if the value is precisely the same ..
This problem doesn't appear for the method `onDuplicateKeyUpdate` with `set` definition.

So I tried to use `onDuplicateKeyUpdate` without `set` definition but .. the SQL built has a wrong syntax

How can I do that?

**Comment by nakamura-to** [comment]
Hello,

Why not get the entity in a separate query?
My idea is as follows:

```kotlin
override suspend fun addMember(guildId: Int, entityId: String): GuildMember {
    requireEntityIdNotBlank(entityId)

    val member = GuildMember(guildId, entityId)

    val query = QueryDsl.insert(_GuildMember.guildMember)
        .onDuplicateKeyIgnore()
        .executeAndGet(member)
    
    val query2 = QueryDsl.from(_GuildMember.guildMember).where { 
        _GuildMember.guildMember.guildId eq member.guildId
        _GuildMember.guildMember.entityId eq member.entityId
    }.single()

    return database.runQuery(query) ?: database.runQuery(query2)
}
```

**Reply by Distractic** [reply]
Thanks for your answer
Because I (always) want to only make a single request for this operation
But if there's no a solution to do that, I will make two requests

**Reply by nakamura-to** [reply, marked as answer]
Thank you for sharing your reason.
I don’t think there is a solution to do that with a single request.
